sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_v6/query.sql

```diff
@@ -1252,7 +1258,7 @@ SELECT
   map_sum_aggregates[
     OFFSET(15)
   ].map AS scalar_parent_browser_ui_interaction_preferences_pane_home_sum,
-  map_sum_aggregates[OFFSET(16)].map AS scalar_parent_urlbar_picked_autofill_sum,
+  map_sum_aggregates[OFFSET(16)].map AS scalar_parent_urlbar_picked_autofill_sum, --WHAT DO WE DO HERE
```

Comment:
This is indeed a pretty fiddly process, and I'd love to see a future where this query is generated from a config file of target metrics, since there's room for subtle mistakes here if ordering gets off.

The reason this creation of arrays and indexing into the arrays exists is a performance optimization so that we can group a bunch of like metrics into a single UDF call. If we didn't build the array, then we'd need to do separate UDF calls and the query would start failing with "query too complex" errors at the planning phase of execution.

What we need to do is to make sure we have metrics listed in exactly the same order in 3 different places:

- pulling the field out of the overall main ping and giving it a "flat" name (order here doesn't actually matter but is good practice to keep this aligned with the following two for readability)
- in the map_sum_aggregates array definition
- indexing into the map_sum_aggregates array

You can either reorder to put these new metrics all at the bottom of their respective lists, in which case you just need to increment the offset on each line. If you insert within the existing lists, then you'll need to renumber all the lines that follow to make sure the offsets exactly match the order of the original array.

Reply:
@relud is very familiar with this process and has been deeply involved in creation and tuning of this `clients_daily` query, so would be a good point of contact going forward for questions on `clients_daily` and `clients_last_seen` machinery.

Reply:
thanks Jeff!